packages/components/src/combobox-control/test/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ComboboxControl from '../';
+
+const timezones = [
+	{ name: 'Greenwich Mean Time', abbreviation: 'GMT' },
+	{ name: 'Universal Coordinated Time', abbreviation: 'UTC' },
+	{ name: 'European Central Time', abbreviation: 'ECT' },
+	{ name: '(Arabic) Egypt Standard Time', abbreviation: 'ART' },
+	{ name: 'Eastern African Time', abbreviation: 'EAT' },
+	{ name: 'Middle East Time', abbreviation: 'MET' },
+	{ name: 'Near East Time', abbreviation: 'NET' },
+	{ name: 'Pakistan Lahore Time', abbreviation: 'PLT' },
+	{ name: 'India Standard Time', abbreviation: 'IST' },
+	{ name: 'Bangladesh Standard Time', abbreviation: 'BST' },
+	{ name: 'Vietnam Standard Time', abbreviation: 'VST' },
+	{ name: 'China Taiwan Time', abbreviation: 'CTT' },
+	{ name: 'Japan Standard Time', abbreviation: 'JST' },
+	{ name: 'Australia Central Time', abbreviation: 'ACT' },
+	{ name: 'Australia Eastern Time', abbreviation: 'AET' },
+	{ name: 'Solomon Standard Time', abbreviation: 'SST' },
+	{ name: 'New Zealand Standard Time', abbreviation: 'NST' },
+	{ name: 'Midway Islands Time', abbreviation: 'MIT' },
+	{ name: 'Hawaii Standard Time', abbreviation: 'HST' },
+	{ name: 'Alaska Standard Time', abbreviation: 'AST' },
+	{ name: 'Pacific Standard Time', abbreviation: 'PST' },
+	{ name: 'Phoenix Standard Time', abbreviation: 'PNT' },
+	{ name: 'Mountain Standard Time', abbreviation: 'MST' },
+	{ name: 'Central Standard Time', abbreviation: 'CST' },
+	{ name: 'Eastern Standard Time', abbreviation: 'EST' },
+	{ name: 'Indiana Eastern Standard Time', abbreviation: 'IET' },
+	{ name: 'Puerto Rico and US Virgin Islands Time', abbreviation: 'PRT' },
+	{ name: 'Canada Newfoundland Time', abbreviation: 'CNT' },
+	{ name: 'Argentina Standard Time', abbreviation: 'AGT' },
+	{ name: 'Brazil Eastern Time', abbreviation: 'BET' },
+	{ name: 'Central African Time', abbreviation: 'CAT' },
+];
+const mapTimezoneOption = ( tz ) => ( {
+	value: tz.abbreviation,
+	label: tz.name,
+} );
+
+const timezoneOptions = timezones.map( mapTimezoneOption );
+const defaultLabelText = 'Select a timezone';
+const getLabel = () => screen.getByText( defaultLabelText );
+const getInput = () => screen.getByRole( 'combobox' );
```

Comment:
Let's also check that the input is labelled correctly:

```suggestion
const getInput = ( name ) => screen.getByRole( 'combobox', { name } );
```

We can then call `getInput( defaultLabelText )` in our tests (or swap `defaultLabelText` with whatever label we need to test for)

Reply:
Good idea, thanks! 👍